### PR TITLE
Fix .offset() and .position() bugs

### PR DESCRIPTION
**Pull request by gibson042**
### Summary ###
Makes `.offset()` always relative to the document and `.position()` always relative to the closest [containing block](https://drafts.csswg.org/css-display-3/#containing-block) offsetParent (if any).

Some new tests fail on IE\<11 (where `document.documentElement.getBoundingClientRect()` erroneously includes margins and disregards viewport scrolling), and have been manually disabled/patched up.

Fixes gh-3080
Fixes gh-3107
Closes gh-3096
Ref gh-3479

### Checklist ###
* [x] All authors have signed the CLA at https://contribute.jquery.com/CLA/
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [x] If needed, a docs issue/PR was created: https://github.com/jquery/api.jquery.com/pull/1027

### CODE REVIEW

File: test/unit/offset.js [hunk 1]
```diff
@@ -503,6 +503,135 @@ QUnit.test( "chaining", function( assert ) {
...
+						// positioning so we don't double its effects.
+						$el.css( { position: "absolute" } );
+					}
+					assert.deepEqual( supportjQuery.extend( {}, $el.position() ), pos,
+						"jQuery('#" + id + "').position() round-trips" );
+
+					// TODO Verify .offset(...)
+					// assert.deepEqual( $el.offset( offset ).offset(), offset )
+					// assert.deepEqual( $el.offset( adjustedOffset ).offset(), adjustedOffset )
+					// assert.deepEqual( $new.offset( offset ).offset(), offset )
```
Comment (gibson042) [inline test/unit/offset.js]: Omitted for scope creep. See gh-3488.

**Comment by gibson042** [comment]
Resolved: There's no need to handle extreme situations in the broken environments.

**Comment by markelog** [comment]
I think we need perf test for it

**Comment by gibson042** [comment]
A perf test for what? I'm not aware of any alternatives here.

**Comment by gibson042** [comment]
@jquery/core This one is finally ready to land. I opened https://github.com/jquery/api.jquery.com/pull/1027 to clarify some details that came to light over the course of working on it.
